bpf/bpf_host.c

```diff
@@ -881,11 +876,6 @@ handle_ipv4_cont(struct __ctx_buff *ctx, __u32 secctx, const bool from_host,
 	}
 
 #if defined(ENABLE_IPSEC) && !defined(TUNNEL_MODE)
-	/* We encrypt host to remote pod packets only if they are from proxy. */
-	if (from_proxy && info->tunnel_endpoint && encrypt_key)
-		return set_ipsec_encrypt(ctx, encrypt_key, info->tunnel_endpoint,
-					 info->sec_identity, true, false);
-
```

Comment:
These packets will then trickle down into `to-netdev` instead, and get encrypted there. Sounds good.

Note that this requires that the BPF programs on native devices are re-loaded first.

Reply:
Yup, I'm about to add a commit which re-implements Paul's original change that forces host device attachment first.